crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -277,24 +277,42 @@ impl UserIdentity {
 ///
 /// Master keys are used to sign other cross signing keys, the self signing and
 /// user signing keys of an user will be signed by their master key.
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, Eq, Serialize, Deserialize)]
 #[serde(try_from = "CrossSigningKey")]
 pub struct MasterPubkey(Arc<CrossSigningKey>);
```

Comment:
Why not just change the `PartialEq` impl of the inner `CrossSigningKey` type? Or is the derived impl on that (intentionally) being used elsewhere?

Reply:
The higher level types guarantee to have the correct `usage`, so the comparison can safely be skipped. Any additional usage in the higher level types doesn't matter for the comparison.

This can't be said for the lower level `CrossSigningKey` type, so I'm reluctant to do `PartialEq` on that type without considering the `usage`.

Reply:
Okay, then `PartialEq` should probably be removed from that type?

Reply:
What's wrong with a `PartialEq` on `CrossSigningKey` that evaluates all fields? Might come in handy at least for some tests.

Reply:
I guess we can drop it since it's not used currently anywhere.

Reply:
> What's wrong with a PartialEq on CrossSigningKey that evaluates all fields? 

Yeah, I'm inclined towards leaving the `PartialEq`/`Eq` there too. I don't think it should just be `PartialEq` since you can compare any two `CrossSigningKey`s.

What I think we need to start doing better is documenting the intended trait impl semantics. For instance, in this case, stating in a doc comment that the intended semantics of `Eq for MasterPubKey` is to check equality of the public keys, ignoring the signatures. Do doc comments on trait impls get rendered?

Reply:
>  I guess we can drop it since it's not used currently anywhere.

I think we raced each other (just saw your comment). I don't feel strongly about it either way so it's fine.

Reply:
That's mostly an internal type so you likely wouldn't look at the rendered version anyways.

Reply:
> What's wrong with a `PartialEq` on `CrossSigningKey` that evaluates all fields?

It becomes easy to accidentally derive `PartialEq` on a new type containing a `CrossSigningKey` and have it use the possibly / likely unwanted `PartialEq` semantics.

Reply:
Right, added some docs and removed the `PartialEq` on the inner type.